vend.js

```diff
@@ -524,11 +524,44 @@ var fetchPaymentTypes = function(args, connectionInfo, retryCounter) {
   return utils.sendRequest(options, args, connectionInfo, fetchPaymentTypes, retryCounter);
 };
 
-var fetchProductTypes = function(args, connectionInfo, retryCounter) {
+var fetchAllVersions = function (args, connectionInfo, retryCounter) {
+
+  //In case if args is not passed while calling function
+  if (args.accessToken && args.domainPrefix) {
+    connectionInfo = args;
+    args = {}; //No args are required for this
+  }
+
+  log.debug('inside fetchAllVersions()');
+  if (!retryCounter) {
+    retryCounter = 0;
+  }
+  else {
+    log.debug('retry #' + JSON.stringify(retryCounter, null, 2));
+  }
+
+  var path = '/api/2.0/versions';
+  var vendUrl = 'https://' + connectionInfo.domainPrefix + '.vendhq.com' + path;
+  var authString = 'Bearer ' + connectionInfo.accessToken;
+  log.debug('GET ' + vendUrl);
+  log.silly('Authorization: ' + authString); // TODO: sensitive data ... do not log?
```

Comment:
you might as well remove this line of code from this code block since you are adding a new feature